ndc-models/src/lib.rs

```diff
@@ -131,9 +136,39 @@ pub enum TypeRepresentation {
     /// Any JSON string
     String,
     /// Any JSON number
+    #[deprecated(since = "0.1.2", note = "Use sized numeric types instead")]
     Number,
     /// Any JSON number, with no decimal part
+    #[deprecated(since = "0.1.2", note = "Use sized numeric types instead")]
     Integer,
+    /// A 8-bit signed integer with a minimum value of -2^7 and a maximum value of 2^7 - 1
+    Int8,
+    /// A 16-bit signed integer with a minimum value of -2^15 and a maximum value of 2^15 - 1
+    Int16,
+    /// A 32-bit signed integer with a minimum value of -2^31 and a maximum value of 2^31 - 1
+    Int32,
+    /// A 64-bit signed integer with a minimum value of -2^63 and a maximum value of 2^63 - 1
+    Int64,
+    /// An IEEE-754 single-precision floating-point number
+    Float32,
+    /// Arbitrary-precision decimal string
+    Decimal,
```

Comment:
nit: You mentioned calling this BigDecimal would be clearer right?

Reply:
yes I'll modify this.